packages/auth/src/core/credentials/email.ts

```diff
@@ -111,13 +113,48 @@ export class EmailAuthCredential extends AuthCredential {
 
   /** @internal */
   async _getIdTokenResponse(auth: AuthInternal): Promise<IdTokenResponse> {
+    async function internalSignInWithPassword(cred: EmailAuthCredential, withRecaptcha: boolean = false): Promise<IdTokenResponse> {
+      let request: SignInWithPasswordRequest;
+      if (withRecaptcha) {
+        const verifier = new RecaptchaEnterpriseVerifier(auth);
+        const captchaResponse = await verifier.verify('signInWithEmailPassword')
+        .catch(() => {
+          return verifier.verify('signInWithEmailPassword', true);
+        }).catch((error) => {
+          return Promise.reject(error);
+        });
+        request = {
+          returnSecureToken: true,
+          email: cred._email,
+          password: cred._password,
+          captchaResponse,
+          clientType: RecaptchaClientType.WEB,
+          recaptchaVersion: RecaptchaVersion.ENTERPRISE,
+        };
+      } else {
+        request = {
+          returnSecureToken: true,
+          email: cred._email,
+          password: cred._password,
+        };
+      }
+      return signInWithPassword(auth, request);
+    }
+
     switch (this.signInMethod) {
       case SignInMethod.EMAIL_PASSWORD:
-        return signInWithPassword(auth, {
-          returnSecureToken: true,
-          email: this._email,
-          password: this._password
-        });
+        if (auth._recaptchaConfig?.emailPasswordEnabled) {
+          return internalSignInWithPassword(this, true);
+        } else {
+          return internalSignInWithPassword(this).catch(async (error) => {
+            if (error.code === `auth/${AuthErrorCode.INVALID_RECAPTCHA_VERSION}`) {
+              console.log("Sign in with email password is protected by reCAPTCHA for this project. Automatically triggers reCAPTCHA flow and restarts the sign in flow.");
```

Comment:
Is this console.log meant to be here?

Reply:
Yeah, the recaptcha flow starts automatically here. So want to inform the developers in case they'r curious about the additional networking calls.